### ISSUE TITLE
Incorrect calling convention result for tail call

### ISSUE THREAD

**Issue by hwu71** [angr/angr#3138]
---

**Describe the bug.**

When the target function is called at the tail position of the caller function, instead of ending with `call addr` instruction, there would be a `jmp addr` instruction at the end of the caller function, pointing to the target function being tail-called.
In this case, the `CallingConventionAnalysis` will fail to identify the actual caller function as the 'caller' of the target function during the process, since `CallingConventionAnalysis` only catches edge with jump kind as `Ijk_Call`. Therefore, the analysis will not enter `self._analyze_callsite()` for this caller function. https://github.com/angr/angr/blob/9ca4fcea39d89e285b47622b21af9b547d3acd44/angr/analyses/calling_convention.py#L244

This results in prototype being the default value: `() -> int`
https://github.com/angr/angr/blob/9ca4fcea39d89e285b47622b21af9b547d3acd44/angr/analyses/calling_convention.py#L220

**Example.**
```
# Caller function #1
FltkImgbuf::FltkImgbuf (Type type, int width, int height)
{
   _MSG("FltkImgbuf: new root %p\n", this);
   init (type, width, height, NULL);
}

# Caller function #2
FltkImgbuf::FltkImgbuf (Type type, int width, int height, FltkImgbuf *root)
{
   _MSG("FltkImgbuf: new scaled %p, root is %p\n", this, root);
   init (type, width, height, root);
}

# Target function, addr: 0x447960
void FltkImgbuf::init (Type type, int width, int height, FltkImgbuf *root)

```
<img src="https://user-images.githubusercontent.com/43049320/154366779-f3dc1ce8-570e-4c66-87ea-9c2eb8c1d9d5.png" width="400" />

**To Reproduce.**

```
import angr
def test_calling_convention(path, addr):
    p = angr.Project(path, load_options={'auto_load_libs': False})
    cfg = p.analyses.CFGFast(normalize=True, show_progressbar=True, data_references=True,
                            force_complete_scan=False)
    func = p.kb.functions.get_by_addr(addr)
    calling_convention = p.analyses.CallingConvention(func=func, cfg=cfg, analyze_callsites=True)
    print("Function %s, prototype: %s" % (hex(addr), calling_convention.prototype))

# dillo-fltk.39-feh
path = "dillo"
addr = 0x447960
test_calling_convention(path, addr)
```
[dillo.zip](https://github.com/angr/angr/files/8084022/dillo.zip)

**Additional context.**

I also tried enabling the `detect_tail_call` option when doing `CFGFast`, the function at address 0x447960 wasn't identified as a tail call, for the reason that: it doesn't meet the requirement in the  `_is_tail_call_optimization()` function that "the destination function is not one of the known functions."

Do you think this issue should be fixed in angr? If yes, could you give me some ideas on how to solve it? Thanks!
@antoniobianchi333

**Comment by ltfish** [comment]
@hwu71 I don't think you are supposed to just assign an issue to me without at least talking to me somewhere, e.g., over Slack.

Talk to me first.
